modules/post/windows/gather/hashdump.rb

```diff
@@ -145,15 +145,16 @@ def capture_hboot_key(bootkey)
         hash.update(vf[0x70, 16] + @sam_qwerty + bootkey + @sam_numeric)
 
         rc4 = OpenSSL::Cipher.new("rc4")
+        rc4.decrypt
         rc4.key = hash.digest
         hbootkey  = rc4.update(vf[0x80, 32])
         hbootkey << rc4.final
         hbootkey
       when 2
         aes = OpenSSL::Cipher.new('aes-128-cbc')
+        aes.decrypt
         aes.key = bootkey
         aes.padding = 0
-        aes.decrypt
         aes.iv = vf[0x78, 16]
         aes.update(vf[0x88, 16]) # we need only 16 bytes
```

Comment:
Example code snippet working in 2.7 but failing in 3:

```ruby
require 'openssl'

boot_key = "A" * 16
value_data = "B" * 2000

aes = OpenSSL::Cipher.new('aes-128-cbc')
aes.key = boot_key
aes.padding = 0
aes.decrypt
aes.iv = value_data[0x78, 16]
aes.update(value_data[0x88, 16])
```

Error:
```ruby
> aes.update(value_data[0x88, 16])
Traceback (most recent call last):
        5: from /usr/local/bin/irb:23:in `<main>'
        4: from /usr/local/bin/irb:23:in `load'
        3: from /usr/local/lib/ruby/gems/3.0.0/gems/irb-1.3.0/exe/irb:11:in `<top (required)>'
        2: from (irb):17:in `<main>'
        1: from (irb):17:in `update'
OpenSSL::Cipher::CipherError (key not set)
```

After moving the decrypt call before the key is set it works as expected:

```ruby
require 'openssl'

boot_key = "A" * 16
value_data = "B" * 2000

aes = OpenSSL::Cipher.new('aes-128-cbc').decrypt
aes.decrypt
aes.key = boot_key
aes.padding = 0
aes.iv = value_data[0x78, 16]
aes.update(value_data[0x88, 16])
```